rust/lance-file/src/page_table.rs

```diff
@@ -36,28 +36,44 @@ pub struct PageTable {
 impl PageTable {
     /// Load [PageTable] from disk.
     ///
-    /// The field_ids that are loaded are `field_id_offset` to `field_id_offset + num_columns`.
-    /// `field_id_offset` should be the smallest field_id in the schema. `num_columns` should
-    /// be the total unique number of field ids, including struct fields despite the fact
-    /// they have no data pages.
+    /// Parameters:
+    ///  * `position`: The start position in the file where the page table is stored.
+    ///  * `min_field_id`: The smallest field_id that is present in the schema.
+    ///  * `max_field_id`: The largest field_id that is present in the schema.
+    ///  * `num_batches`: The number of batches in the file.
+    ///
+    /// The page table is stored as an array. The on-disk size is determined based
+    /// on the `min_field_id`, `max_field_id`, and `num_batches` parameters. If
+    /// these are incorrect, the page table will not be read correctly.
```

Comment:
Given only `min` and `max` how do we handle holes?  Or is a "hole" only at the beginning or end (and all fields within a file are contiguous)

Reply:
Holes are written to the page table as empty pages, just like struct fields. (This is one of the things that makes it very inefficient. If you have field ids `0, 50`, it has to store empty page data for all the non-existent fields `1..49`.)  I can make this explicit in the docs.